scripts/dqlite/cmd/main.go

```diff
@@ -183,6 +200,42 @@ func main() {
 	<-ctx.Done()
 }
 
+func dumpDB(ctx context.Context, dial client.DialFunc, db *sql.DB, address, dir string) error {
+	cli, err := client.New(ctx, address, client.WithDialFunc(dial))
+	if err != nil {
+		return fmt.Errorf("client.New failed %w", err)
+	}
+
+	if dir == "" {
+		var err error
+		dir, err = os.Getwd()
+		if err != nil {
+			return fmt.Errorf("os.Getwd() failed %w", err)
+		}
+		dir = filepath.Join(dir, "dump")
+
+	}
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("os.MkdirAll() failed %w", err)
+	}
+
+	files, err := cli.Dump(ctx, "tmp")
+	if err != nil {
+		return fmt.Errorf("dump failed")
+	}
+
+	for _, file := range files {
+		path := filepath.Join(dir, file.Name)
+
+		err := os.WriteFile(path, file.Data, 0600)
```

Comment:
This certainly seems to be something where dqlite assumes everything can be held in memory, vs a io.Reader being passed here. That said, if this is the interface they give us, then we'll use it.
We might at least recommend that it would be better if this was designed to stream content for the future.

Reply:
So the fun part of this, is that this is getting sent back over the Dqlite server. So if you have 1TB of data, guess we'll be sending over 1TB of data.

We definitely should have something better.